File: Sample Code/ObjcSampleCode/DJISdkDemo/Demo/DJIMapView/DJIAnnotationViews/DJIAircraftAnnotation.m
//
//  DJIAircraftAnnotation.m
//  Phantom3
//
//  Created by Ares on 14-8-21.
//  Copyright (c) 2014年 Jerome.zhang. All rights reserved.
//

#import "DJIAircraftAnnotation.h"

@implementation DJIAircraftAnnotation
{
    NSString* _annotationTitle;
}

+ (instancetype)annotationWithCoordinate:(CLLocationCoordinate2D)coordinate heading:(CGFloat)heading
{
    DJIAircraftAnnotation *instance = [self annotationWithCoordinate:coordinate];
    if (instance) {
        instance.heading = heading;
    }
    
    return instance;
}

-(id) initWithCoordinate:(CLLocationCoordinate2D)coordinate
{
    self = [super init];
    if (self) {
        self.coordinate = coordinate;
    }
    
    return self;
}

-(id) init
{
    self = [super init];
    if (self) {
        _annotationTitle = nil;
    }
    
    return self;
}

-(void) setAnnotationTitle:(NSString*)title
{
    _annotationTitle = title;
}

-(NSString*) title
{
    return _annotationTitle;
}

@end

File: Sample Code/SwiftSampleCode/DJISDKSwiftDemo/Missions/HotPoint/DJIAircraftAnnotation.swift
//
//  DJIAircraftAnnotation.h
//
//  Created by DJI on 14-8-21.
//
import CoreLocation
import CoreGraphics

class DJIAircraftAnnotation: DJIAnnotation {
    var heading:Double?=nil
    var annotationTitle:String?=nil
   
 
    init(coordinate: CLLocationCoordinate2D, heading:Double) {
        self.heading = heading;
        super.init(coordinate: coordinate)
    }
    

    func title() -> String {
        return annotationTitle!
    }
}
